basculechecks/metricvalidator_test.go

```diff
@@ -374,14 +388,22 @@ func TestPrepMetrics(t *testing.T) {
 				require.Nil(err)
 				auth.Request.URL = u
 			}
+			if tc.includeMethod {
+				auth.Request.Method = "get"
+			}
 
-			c, partner, endpoint, reason, err := m.prepMetrics(auth)
-			if tc.includeToken {
+			c, partner, endpoint, err := m.prepMetrics(auth)
+			if tc.includeToken && tc.includeMethod {
 				assert.Equal(client, c)
 			}
 			assert.Equal(tc.expectedPartner, partner)
 			assert.Equal(tc.expectedEndpoint, endpoint)
-			assert.Equal(tc.expectedReason, reason)
+			if tc.expectedReason != "" {
+				var r Reasoner
+				if assert.True(errors.As(err, &r)) {
+					assert.Equal(tc.expectedReason, r.Reason())
```

Comment:
If you always return sentinel, you could use `errors.Is` for assertion in line 410 😎